indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -0,0 +1,122 @@
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_template
+
+
+def _get_session_list_args(event):
+    return {'sessions': get_active_sessions(event), 'default_colors': get_colors()}
+
+
+def _render_session_list(event):
+    tpl = get_template_module('events/sessions/management/_session_list.html')
+    return tpl.render_session_list(event, **_get_session_list_args(event))
+
+
+class RHSessionsList(RHManageSessionsBase):
+    """Display list of all sessions within the event"""
+
+    def _process(self):
+        return WPManageSessions.render_template('management/session_list.html', self._conf,
+                                                event=self.event_new, **_get_session_list_args(self.event_new))
+
+
+class RHCreateSession(RHManageSessionsBase):
+    """Create a session in the event"""
+
+    def _get_random_color(self):
+        used_colors = {s.colors for s in self.event_new.sessions.filter_by(is_deleted=False)}
+        unused_colors = set(get_colors()) - used_colors
+        return random.choice(tuple(unused_colors) or get_colors())
+
+    def _process(self):
+        form = SessionForm(obj=FormDefaults(colors=self._get_random_color()))
+        if form.validate_on_submit():
+            create_session(self.event_new, form.data)
+            return jsonify_data(html=_render_session_list(self.event_new))
+        return jsonify_template('events/sessions/management/add_session.html', form=form)
+
+
+class RHModifySession(RHManageSessionBase):
+    """Modify a session"""
+
+    def _process(self):
+        form = SessionForm(obj=self.session)
+        if form.validate_on_submit():
+            update_session(self.session, form.data)
+            return jsonify_data(html=_render_session_list(self.event_new))
+        return jsonify_template('events/sessions/management/add_session.html', form=form)
+
+
+class RHDeleteSessions(RHManageSessionsBase):
+    """Remove multiple sessions"""
+
+    def _process(self):
+        session_ids = map(int, request.form.getlist('session_id'))
+        sessions = self.event_new.sessions.filter(Session.id.in_(session_ids), ~Session.is_deleted)
+        for sess in sessions:
+            delete_session(sess)
+        return jsonify_data(html=_render_session_list(self.event_new))
+
+
+class RHExportSessionsCSV(RHManageSessionsBase):
+    """Export list of sessions to a CSV"""
+
+    def _process(self):
+        pass
+
+
+class RHExportSessionsPDF(RHManageSessionsBase):
+    """Export list of sessions to a PDF"""
+
+    def _process(self):
+        pass
+
+
+class RHSessionREST(RHManageSessionBase):
+    """Perform update or removal of a session"""
+
+    def _process_DELETE(self):
+        delete_session(self.session)
+        return jsonify_data(html=_render_session_list(self.event_new))
+
+    def _process_PATCH(self):
+        data = request.json
+        if data.viewkeys() > {'colors'}:
+            raise BadRequest
+        if 'colors' in data:
```

Comment:
I'd feel safer with something like this:

```
updates = {}
if 'colors' in data:
    ...
    updates['colors'] = ...
update_session(self.session, updates)
```

But we could simply change it once we add more PATCHable fields here